File: combo2-advanced-agentic-engineering/tiny-agent-csharp/tests/TinyAgent.Tests/ToolsFactory.cs
using TinyAgent.Shared;

namespace TinyAgent.Tests;

/// <summary>
/// The tool implementation under test. See <see cref="Impl"/> for the switch.
/// </summary>
public static class ToolsFactory
{
    public static ITools Create(string sandboxRoot) =>
        Impl.Selected() == "reference"
            ? new Reference.ReferenceTools(sandboxRoot)
            : new Starter.StarterTools(sandboxRoot);

    public static string Describe() =>
        Impl.Selected() == "reference"
            ? "reference (worked solution)"
            : "starter (your code)";
}

File: combo2-advanced-agentic-engineering/tiny-agent-java/tests/src/test/java/ai/octoco/tinyagent/tests/ToolsFactory.java
package ai.octoco.tinyagent.tests;

import ai.octoco.tinyagent.reference.ReferenceTools;
import ai.octoco.tinyagent.shared.Tools;
import ai.octoco.tinyagent.starter.StarterTools;

/** The tool implementation under test. See {@link Impl} for the switch. */
final class ToolsFactory {

    private ToolsFactory() {}

    static Tools create(String sandboxRoot) {
        return "reference".equals(Impl.selected())
                ? new ReferenceTools(sandboxRoot)
                : new StarterTools(sandboxRoot);
    }

    static String describe() {
        return "reference".equals(Impl.selected()) ? "reference (worked solution)" : "starter (your code)";
    }
}
